src/sentry/static/sentry/app/components/activity/note.jsx

```diff
@@ -6,7 +6,52 @@ import ConfigStore from '../../stores/configStore';
 import LinkWithConfirmation from '../../components/linkWithConfirmation';
 import {t} from '../../locale';
 
+function isSafeHref(href) {
+  try {
+    return /^https?:\/\/./i.test(decodeURIComponent(unescape(href)));
+  } catch(e) {
+    return false;
+  }
+}
+
+
+// We need to implement our own marked Renderer to not render
+// potentially malicious uris.
+// This is copy/pasted from
+// https://github.com/chjj/marked/blob/master/lib/marked.js#L869-L888
+// and modified.
+function Renderer(options) {
+  this.options = options || {};
+}
+
+Object.assign(Renderer.prototype, marked.Renderer.prototype);
+
+Renderer.prototype.link = function(href, title, text) {
+  // For a bad link, just return the plain text href
+  if (this.options.sanitize && !isSafeHref(href)) return href;
```

Comment:
> Then do this. Wrap link, and if the output is '', output href instead. Again, I think this can be implemented with much less code and be more compatible with Marked.

I don't like this because that means it gets looped over again with useless logic since we are intending to change the behavior, so we're hitting the old behavior again.

I agree it's not ideal, but I also don't care about compatibility. They also fully document swapping in a custom `renderer`, so I don't see the issue. It just spits out some HTML.